### ISSUE TITLE
Libraries trimming test are missing a lot of configurations

### ISSUE THREAD

**Issue by marek-safar** [dotnet/runtime#39274]
Newly added trimming libraries tests are running only for configurations which are not by default trimmed aggressively. All the ones which are missing so we have no test coverage.

Mandatory RIDs to be added to the existing matrix
- browser-wasm
- ios-arm64
- tvos-arm64
- android-arm64

**Comment by safern** [comment]
Cc: @joperezr

**Comment by joperezr** [comment]
Although we do plan to enable a few extra platforms, with the trimming tests we are manually changing the way we invoke the linker in order to force aggressive trimming so we do have coverage for now. That said, I do want to restate that we do plan to add at least wasm platform as well

**Comment by safern** [comment]
@joperezr could you add the right milestone on when you're going to be adding these? At least browser seems important for 5.0.0

**Comment by joperezr** [comment]
yup, wasm will be done before 5.0 definitely.

**Comment by ViktorHofer** [comment]
@joperezr it's unclear to me what the remaining work is. Is someone actively working on this? Thanks

**Comment by joperezr** [comment]
We still haven't added a wasm leg for the linker tests, today we only have these 3:

https://github.com/dotnet/runtime/blob/afa4b2f344192c3609dd3c1419f68d99d10d7d62/eng/pipelines/runtime-linker-tests.yml#L61-L64

Ideally we would want to add wasm in there as well to build that vertical and run linker tests for that runtime, so that would be the remaining work. @eerhardt do we still want to do that for 5.0?

**Comment by eerhardt** [comment]
> Ideally we would want to add wasm in there as well to build that vertical and run linker tests for that runtime, so that would be the remaining work. @eerhardt do we still want to do that for 5.0?

It would be good to get the wasm leg in, but I don't think it is required for 5.0 at this point. It can be done early in 6.0.

**Comment by danmosemsft** [comment]
Added to story https://github.com/dotnet/runtime/issues/43078

**Comment by marek-safar** [comment]
I think this is ready from our side. Running the linker against listed RIDs should be possible a long time ago and we now have AndroidAppBuilderTask, AppleAppBuilderTask and WasmBuildApp to produce the executable and Helix integration to run it.

**Comment by danmosemsft** [comment]
@eerhardt could you summarize the current thinking here -- I believe you talked to @marek-safar -- what are the right next actions here do you see?

**Comment by eerhardt** [comment]
We will start off by adding a wasm leg, and then ios and android legs to the trimming tests.

**Comment by danmosemsft** [comment]
sounds good. To summarize my understanding,
* we don't need Xamarin/Blazor libraries/templates/SDK support here; that kind of testing has to happen upstack, both functionally, and size (latter in perf repo)
* we need to publish for their RIDs as that swaps in different implementations of our libraries in some cases
* the tests verify that (a) there are no linker errors/warnings (b) libraries functionality doesn't break under trimming 
* we write these tests to focus on particular "tricky" API we have annotated -- eg., where there is serialization - to ensure our annotations remain good

**Comment by eerhardt** [comment]
That summary looks good except:

> the tests verify that (a) there are no linker errors/warnings (b) libraries functionality doesn't break under trimming

Technically (a) happens during the build, and not during the tests. The tests verify "(b) libraries functionality doesn't break under trimming".

**Comment by danmosemsft** [comment]
Ah, saw warningsaserrors on the test project build and assumed it was for that purpose. Fine!

**Comment by eerhardt** [comment]
Moving to 7.0. We added a browser-wasm leg, which covers the basic mono scenarios. We should enable android and ios in 7.0.

**Comment by steveisok** [comment]
This is something my team can help out on.  These tests seem similar to our functional tests and I don't think would take too much effort to enable.

**Comment by carlossanlop** [comment]
@joperezr @steveisok should we move this to 8.0?

**Comment by joperezr** [comment]
I'll defer to @steveisok on this. Ideally we would want to have these tests running on all different runtimes that we cross-target, but the most important thing here is to decide when do we want to do that work.

**Comment by ViktorHofer** [comment]
ping @steveisok for the above question

**Comment by steveisok** [comment]
Let's leave it in 7.  I could end up being wrong, but I believe the work is hooking up to our mobile testing targets the same way the functional tests (src/tests/FunctionalTests) do. If I'm correct, then I think it's less than a few days of work at worst.

**Comment by ericstj** [comment]
@steveisok - should we keep this in 7.0 or move it out at this point?

**Comment by directhex** [comment]
Are we good on this for wasm, i.e. was #48429 everything needed? If so, I can duplicate that PR's changes for iOS, tvOS and Android

**Comment by steveisok** [comment]
> Are we good on this for wasm, i.e. was #48429 everything needed? If so, I can duplicate that PR's changes for iOS, tvOS and Android

As far as I know.  The wiring up of these tests for mobile should be the same as the functional tests.
